test/integration/targets/ec2_vpc_nat_gateway/tasks/main.yml

```diff
@@ -1,82 +1,212 @@
+  - name: create subnet 
+    ec2_vpc_subnet:
+      cidr: "10.232.232.128/28"
+      vpc_id: "{{ vpc_result.vpc.id }}"
+      <<: *aws_connection_info
+      state: present
+    register: vpc_subnet_created
 
-- name: Create new nat gateway with eip allocation-id
-  ec2_vpc_nat_gateway:
-    subnet_id: subnet-12345678
-    allocation_id: eipalloc-12345678
-    wait: yes
-    region: us-west-2
-  register: nat_gateway
-  check_mode: yes
+  # ============================================================
+  - name: create NAT gateway (expected changed=true)
+    ec2_vpc_nat_gateway:
+      state: present
+      subnet_id: "{{ vpc_subnet_created.subnet.id }}"
+      if_exist_do_not_create: 'no'
+      <<: *aws_connection_info
+    register: vpc_nat_gateway_created
+  
+  - name: assert creation happened (expected changed=true)
+    assert:
+      that:
+          - 'vpc_nat_gateway_created'
+          - 'vpc_nat_gateway_created.nat_gateway_id.startswith("nat-")'
+          - 'vpc_nat_gateway_created.vpc_id == vpc_result.vpc.id'
+          - 'vpc_nat_gateway_created.tags == {}'
+  
+  # ============================================================
+  - name: attempt to recreate nat_gateway with if_exist_do_not_create=true (expected changed=false)
+    ec2_vpc_nat_gateway:
+      state: present
+      subnet_id: "{{ vpc_subnet_created.subnet.id }}"
+      <<: *aws_connection_info
+      if_exist_do_not_create: 'yes'
+    register: vpc_nat_gateway_recreated
+  
+  - name: assert recreation did nothing (expected changed=false)
+    assert:
+      that:
+          - 'vpc_nat_gateway_recreated.changed == False'
+          - 'vpc_nat_gateway_recreated.nat_gateway_id == vpc_nat_gateway_created.nat_gateway_id'
+          - 'vpc_nat_gateway_recreated.subnet_id == vpc_nat_gateway_created.subnet_id'
 
-- assert:
-    that:
-      - nat_gateway.changed
+  # ============================================================
+  - name: Add tags to nat-gateway
+    ec2_vpc_nat_gateway:
+      if_exist_do_not_create: 'yes'
+      tags:
+        Name: "{{ resource_prefix }}-nat"
+      subnet_id: "{{ vpc_subnet_created.subnet.id }}"
+      nat_gateway_id: "{{ vpc_nat_gateway_created.nat_gateway_id }}"
+      <<: *aws_connection_info
+    register: vpc_nat_gateway_tagged
+  
+  - name: assert tag is added
+    assert:
+      that:
+          - 'vpc_nat_gateway_tagged.changed == True'
+  
+  # ============================================================
+  - name: Collect nat-gateway facts to check tags
+    ec2_vpc_nat_gateway_info:
+      nat_gateway_ids: "{{ vpc_nat_gateway_created.nat_gateway_id }}"
+      <<: *aws_connection_info
+    register: vpc_nat_gateway_facts
+    until: vpc_nat_gateway_facts | json_query('result[].tags[].Name') | count == 1
+    retries: 10
+    delay: 5
+  
+  - name: assert tag is added
+    assert:
+      that:
+          - 'vpc_nat_gateway_facts.result.0.tags.Name == "{{ resource_prefix }}-nat"'
```

Comment:
It'd be good to add a test for ec2_vpc_nat_gateway_info using filters

```
  - name: Collect nat-gateway facts to check tags
    ec2_vpc_nat_gateway_info:
      filters:
        "tag:Name": "{{ resource_prefix }}-nat"
    register: vpc_nat_gateway_facts

  - name: assert tag is added
    assert:
      that:
          - 'vpc_nat_gateway_facts.result.0.tags.Name == "{{ resource_prefix }}-nat"'
```

Once you've done that it'd be good to add ec2_vpc_nat_gateway_info to  test/integration/targets/ec2_vpc_nat_gateway/aliases